processor/transform/embeds.ts

```diff
@@ -0,0 +1,27 @@
+import { visit } from 'unist-util-visit';
+
+import { NodeTypes } from '../../enums';
+
+const embedTransformer = () => {
+  return (tree: any) => {
+    visit(tree, 'link', (node, _, parent) => {
+      try {
+        if (parent.children.length > 1 || node.title !== '@embed') return;
```

Comment:
Perhaps we should check if the parent is a paragraph, and completely replace it?

Reply:
yeah i should still be checking if the parent is a paragraph, i think i uh optimized that out at one point, heh. but i'm basically just reforming the node and replacing the parent with it--should i be splicing it in and out i or something nstead?